[clarabellelim] iP

clarabellelim opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

clarabellelim committed:
Level 0

clarabellelim committed:
Level 1. Echo

clarabellelim committed:
Level 2. Add, List

clarabellelim committed:
Level 3. Mark as Done

clarabellelim committed:
Level 4. ToDos, Events, Deadlines

clarabellelim committed:
Duke A-TextUiTesting: Automated Text UI Testing

clarabellelim committed:
Level 5. Handle Errors

clarabellelim committed:
Level 6. Delete

clarabellelim committed:
add enum

clarabellelim committed:
Level 7. Save

clarabellelim committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Level 7. Save

clarabellelim committed:
Level 8. Dates and Times

clarabellelim committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Level 8. Dates and Times

kokseen1 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,38 @@
+import java.io.*;
```
Imported classes should always be listed explicitly!

kokseen1 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,40 @@
+
+    public static Task parseTask(String line) throws GhostException {
+        String[] parts = line.split(" \\| ");
+        return switch (parts[0]) {
+            case "T" -> new Todo(parts[2]);
+            case "D" -> new Deadline(parts[2], parts[3]);
+            case "E" -> new Event(parts[2], parts[3], parts[4]);
+            default -> throw new GhostException("Unknown task type.");
```
According to the coding standard, `case` clauses should not have indentation.

kokseen1 reviewed (commented):
Hi! You have good code that complies closely with coding standards! Perhaps consider organizing some classes into packages, and add some JavaDoc for classes and methods. That's all!

clarabellelim committed:
A-MoreOOP

clarabellelim committed:
A-Packages

clarabellelim committed:
Divide classes into packages

clarabellelim committed:
A-Packages

clarabellelim committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

geeliette commented on src/main/java/ghost/parser/Parser.java:
```diff
@@ -0,0 +1,74 @@
+            case "todo", "deadline", "event" -> new AddCommand(input);
+            default -> throw new GhostException(" AHHHHHH: The description is too scary, I can't understand it!");
+        };
+    }
```
Remember that there are no indentations for case clauses for this course!

geeliette commented on src/main/java/ghost/task/TaskList.java:
```diff
@@ -0,0 +1,117 @@
+import java.util.ArrayList;
+
+public class TaskList {
+    private final ArrayList<Task> tasks;
```
Nice job on using plural form for representing a list of tasks :)

geeliette commented on src/main/java/ghost/parser/Parser.java:
```diff
@@ -0,0 +1,74 @@
+                }
+            }
+            case "finddate" -> {
+                if (parts.length < 2) throw new GhostException("AHHHHHH: Please specify the haunting date in yyyy-MM-dd format.");
```
I like how you give clear instructions on how the date format should be like to users

geeliette commented on src/main/java/ghost/command/ExitCommand.java:
```diff
@@ -0,0 +1,18 @@
+
+public class ExitCommand extends Command {
+    @Override
+    public boolean execute(TaskList tasks, Ui ui, Storage storage) {
```
As this is a boolean method, should it be named to sound like booleans?

geeliette commented on src/main/java/ghost/command/ExitCommand.java:
```diff
@@ -0,0 +1,18 @@
+    }
+
+    @Override
+    public boolean isExit() {
```
I think this is a good example of a boolean method having an appropriate name :)

geeliette reviewed (commented):
LGTM! I like how all the responses are unique and ghost-related. The code is also easy to read with minor issues with regards to the course's java coding standard.

Remember to add javadoc comments!

clarabellelim committed:
A-Gradle

clarabellelim committed:
A-JUnit

clarabellelim committed:
A-JavaDoc: Add JavaDoc comments to the code.

clarabellelim committed:
A-CodingStandard

clarabellelim committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  A-CodingStandard

# Conflicts:
#	src/main/java/ghost/task/Deadline.java
#	src/main/java/ghost/task/Event.java

clarabellelim committed:
Level 9. Find

clarabellelim committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Level 9. Find

# Conflicts:
#	src/main/java/ghost/ui/Ui.java

clarabellelim committed:
Add JavaDoc comments to Level-9: Find.

clarabellelim committed:
Update README.md

clarabellelim committed:
Added main file and fixed jar file errors

clarabellelim committed:
ui: Use 'keyword' parameter in showFindMessage

The 'keyword' parameter in Ui's showFindMessage method is currently
unused, leading to a warning. This reduces code clarity and may
confuse those viewing the code.

Let's update showFindMessage to display the searched keyword in
its output, making the message more informative.

clarabellelim committed:
parser: Fix missing import for FindByKeywordCommand

The `FindByKeywordCommand` class is not recognised in `Parser.java`
due to a missing import statement, causing a "Cannot resolve symbol"
compilation error.

Adding the missing `import ghost.command.FindByKeywordCommand;`
resolves this issue, allowing the parser to recognise and execute
the "find" command correctly.

Let's add the import statement to fix this issue.

clarabellelim committed:
Fix: Refactor Task class to handle user input commands correctly

- Updated `Task.fromString()` to parse saved task formats only.
- Added `Task.fromUserInput()` to handle user input commands like "todo", "deadline", and "event".
- Fixed `AddCommand` to use `Task.fromUserInput()` for creating tasks from user input.
- Corrected parsing logic for `deadline` and `event` tasks to handle date formats properly.

clarabellelim committed:
fix: correct index handling for delete command

The delete command was recognising the first task as index 2 due to
improper handling of the 1-based index provided by the user. This
issue occurred because the provided index was not adjusted correctly
for the internal 0-based index system.

Let's:
* adjust the 1-based index input by subtracting 1 to match the internal
  0-based index for task deletion
* ensure that the task deletion logic works as expected for all valid
  task indices

This change ensures that the delete command works with the correct
task index and provides a user-friendly experience.

clarabellelim committed:
docs: update README to reflect new features and improvements

The README has been updated to better describe the current features
and usage of the GhostTask application. The following changes have
been made:

- Rebranded the project name from "Ghost" to "GhostTask"
- Updated the key features to include task management, deadline, and
  event tracking
- Enhanced the "How to Use" section to include new functionalities
  such as filtering tasks by date and keyword
- Improved the "Tasks to do" list to reflect the current progress and
  upcoming features
- Added a new section with a spooky and fun tone to describe the
  project more engagingly

Let's:
* improve the clarity of the user instructions
* align the feature descriptions with the current project
  functionalities
* update the roadmap to reflect the current progress and the upcoming
  features

clarabellelim committed:
Update README.md

docs: update README to reflect new features and improvements

The README has been updated to better describe the current features 
and usage of the GhostTask application. The following changes have 
been made:

- Rebranded the project name from "Ghost" to "GhostTask"
- Updated the key features to include task management, deadline, and 
  event tracking
- Enhanced the "How to Use" section to include new functionalities
  such as filtering tasks by date and keyword
- Improved the "Tasks to do" list to reflect the current progress and
  upcoming features
- Added a new section with a spooky and fun tone to describe the 
  project more engagingly

Let's:
* improve the clarity of the user instructions
* align the feature descriptions with the current project 
  functionalities
* update the roadmap to reflect the current progress and upcoming 
  features

clarabellelim committed:
Level-10. Add a GUI to the chatbot. Use the JavaFX technology to implement the GUI.

clarabellelim committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Level-10. Add a GUI to the chatbot. Use the JavaFX technology to implement the GUI.

# Conflicts:
#	src/main/java/ghost/command/AddCommand.java
#	src/main/java/ghost/command/DeleteCommand.java
#	src/main/java/ghost/parser/Parser.java
#	src/main/java/ghost/task/Task.java
#	src/main/java/ghost/ui/Ui.java

clarabellelim committed:
edited AddCommand to for gui

clarabellelim committed:
Jar 4.0

clarabellelim committed:
A-UserGuide: Add a User Guide

clarabellelim committed:
remove double readme files

clarabellelim committed:
fix(build): Set mainClass to Launcher in build.gradle

The mainClass in build.gradle was not pointing to the correct class,
causing issues when running the application. Updated it to point to
Launcher to ensure proper execution.

clarabellelim committed:
refactor(code): Replace statement lambda with expression lambda

Replaced statement lambda with expression lambda to improve readability
and conciseness.

clarabellelim committed:
refactor(code): Replace statement lambda with expression lambda

Replaced statement lambda with expression lambda to improve readability
and conciseness.

clarabellelim committed:
refactor(switch): Use enhanced switch for conciseness and readability

Replaced traditional switch statement with enhanced switch expression.

- More concise: Removes the need for break statements.
- Improved readability: Uses `->` instead of case blocks.
- Ensures proper value return: Uses `yield` instead of `return` in block-style cases.

clarabellelim committed:
docs(javadoc): Update Javadoc comments for clarity and consistency

Improved Javadoc comments across the codebase for better clarity and
consistency, ensuring they align with the coding standards.

clarabellelim committed:
fix(build): Set mainClass to Launcher in build.gradle

The mainClass in build.gradle was not pointing to the correct class,
causing issues when running the application. Updated it to point to
Launcher to ensure proper execution.
